### PR TITLE
Feature/upgrade to nx 20

### LINKED ISSUE #116: Executor fails for nx 20
### If possible, please provide a reproduction repository you discovered this bug in.

_No response_

### Which area/package is the issue in?

@koliveira15/nx-sonarqube

### Description

Currently, migrating to nx 20 will cause nx-sonarqube's executor to fail to run.

### If applicable, please provide the Nx report

```bash
Node           : 22.9.0
OS             : darwin-arm64
Native Target  : aarch64-macos
pnpm           : 9.12.1

nx (global)        : 20.0.0
nx                 : 20.0.0
@nx/js             : 20.0.0
@nx/jest           : 20.0.0
@nx/eslint         : 20.0.0
@nx/workspace      : 20.0.0
@nx/cypress        : 20.0.0
@nx/eslint-plugin  : 20.0.0
@nx/next           : 20.0.0
@nx/react          : 20.0.0
@nx/storybook      : 20.0.0
typescript         : 5.6.3
---------------------------------------
Registered Plugins:
@nx/cypress/plugin
@nx/next/plugin
@nx/eslint/plugin
---------------------------------------
Community plugins:
@koliveira15/nx-sonarqube : 4.1.0
nx-stylelint              : 18.0.0-beta.4

```

### Please provide the exception or error you saw

```true
The SonarQube scan failed for project '[REDACTED]'
TypeError: Cannot read properties of undefined (reading 'projects')
    at [REDACTED]/node_modules/.pnpm/@koliveira15+nx-sonarqube@4.1.0_@nx+devkit@20.0.0_nx@20.0.0_@swc-node+register@1.10.9_@swc+co_ak2va7zieaom4bi7d4q6krshfu/node_modules/@koliveira15/nx-sonarqube/src/executors/scan/utils/utils.js:70:56
    at Generator.next (<anonymous>)
    at fulfilled ([REDACTED]/node_modules/.pnpm/tslib@2.7.0/node_modules/tslib/tslib.js:166:62)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
```

### Other information

I tracked the issue to the following:
https://github.com/koliveira15/nx-sonarqube/blob/991ee6b20551bad29381a24099c517b6c72d92a9/packages/nx-sonarqube/src/executors/scan/utils/utils.ts#L96C32-L96C70

```typescript
const projectConfiguration = context.workspace.projects[context.projectName];
```

`context.workspace` is undefined

i'm willing to submit a PR, however I would need to take time to dig into the issue - my guess is that nx-sonarqube needs to be updated to support nx 20

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by gcko**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [X] Configuration related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [X] @koliveira15/nx-sonarqube
- [X] docs-site (A fix to the `build-docs` command)

## What is the current behavior?

Currently, nx-sonarqube fails when using NX 20 

Closes #116 

This also fixes an issue with tsquery and latest typescript. See this bug https://github.com/phenomnomnominal/tsquery/issues/98

## What is the new behavior?

Updated the usages of `workspace` to `projectConfigurations`

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

It has been tested to work correctly with NX 20 and NX 19. I have not tested with other versions of NX

### CODE REVIEW

**Comment by gcko** [comment]
hey @koliveira15 , let me know if there is anything I can do to help out with this PR - 🙏

File: nx.json [hunk 1]
```diff
@@ -46,5 +46,6 @@
   "nxCloudAccessToken": "MWM4ZWZiNDgtOGYwMi00ZDI3LTllMDUtYTNkYTExMmU3MDY2fHJlYWQ=",
   "parallel": 3,
   "defaultBase": "origin/main",
-  "useInferencePlugins": false
+  "useInferencePlugins": false,
+  "useLegacyCache": true
```
Comment (Tommy228) [inline nx.json:50]: This may come from the automatic migrations, but using the new DB cache shouldn't be an issue for this project right?

**Comment by jaredterm** [comment]
Hey @koliveira15 , please let me know if there is anything else needed for this PR.

As an aside, it might be worthwhile contacting NX and making this an official plugin 🤔 IMO static analysis is integral to development pipeline and I personally would push strongly for its inclusion to the NX family

**Comment by gcko** [comment]
@koliveira15 Anything I can do to help? This is becoming a blocker in our repository as it precludes us from updating `nx` and associated dependencies. Please let me know how I can contribute 🙏

**Comment by ErickRodrCodes** [comment]
since he is busy, I will fork it out and move it to 20. we are having the same blocker as it is critical for us to use `nx import` to speed up monorepo work on our side.

**Comment by ErickRodrCodes** [comment]
It seems I am unable to create the package as I'm tied to several things at work. If there is no update in 15 days, I will retry to fork it and somehow own it.

File: migrations.json
Comment (koliveira15) [inline migrations.json:1]: add `migrations.json` so it's easier to track down what was automated in this upgrade

**Review by koliveira15 — CHANGES_REQUESTED** [review]

File: migrations.json
Comment (gcko) [reply migrations.json:1]: @koliveira15 done, I also updated nx to the latest and bumped some of the other minor dependencies as well 🙏

**Review by koliveira15 — APPROVED** [review]

**Pull request merged**